### PR TITLE
[MIG] base_duplicate_security_group: Migration to 13.0

### PR DESCRIPTION
**Pull request by chienandalu**
This module adds a security group to restrict which users can make use of the duplicate
records function or not.

cc @Tecnativa TT30941

please review @Tardo @CarlosRoca13

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]
Code review. Not possible to try on runbot as not available.

**Review by CarlosRoca13 — APPROVED** [review]
LGTM

File: base_duplicate_security_group/static/src/js/duplicate.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    const DuplicateViewMixin = {
+        init: function() {
+            this._super.apply(this, arguments);
+            const _this = this;
+            const base_group = "base_duplicate_security_group.group_duplicate_records";
+            session.user_has_group(base_group).then(function(result) {
+                if (!result) {
+                    _this.controllerParams.activeActions.duplicate = false;
+                }
+            });
```
Comment (Tardo) [inline base_duplicate_security_group/static/src/js/duplicate.js]: Can use arrow functions to preserve the context.
```suggestion
            const base_group = "base_duplicate_security_group.group_duplicate_records";
            session.user_has_group(base_group).then(result => {
                if (!result) {
                    this.controllerParams.activeActions.duplicate = false;
                }
            });
```

File: base_duplicate_security_group/views/assets.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                type="text/javascript"
+                src="/base_duplicate_security_group/static/src/js/duplicate.js"
+            />
+        </xpath>
+    </template>
+    <template
+        id="assets_common"
+        name="Base Duplicate Security Group Assets Tests"
+        inherit_id="web.assets_common"
+    >
```
Comment (Tardo) [inline base_duplicate_security_group/views/assets.xml]:
```suggestion
    <template
        id="assets_tests"
        name="Base Duplicate Security Group Assets Tests"
        inherit_id="web.assets_tests"
    >
```

File: base_duplicate_security_group/static/src/js/duplicate.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        },
+    };
+
+    FormView.include({
+        init: function() {
+            this._super.apply(this, arguments);
+            DuplicateViewMixin.init.apply(this, arguments);
+        },
+    });
+});
```
Comment (Tardo) [inline base_duplicate_security_group/static/src/js/duplicate.js:29]:
```suggestion

    return DuplicateViewMixin;
});
```

File: base_duplicate_security_group/static/src/js/duplicate.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            });
+        },
+    };
+
+    FormView.include({
+        init: function() {
+            this._super.apply(this, arguments);
+            DuplicateViewMixin.init.apply(this, arguments);
+        },
+    });
```
Comment (Tardo) [inline base_duplicate_security_group/static/src/js/duplicate.js]: This doesn't work?
```suggestion
    FormView.include(DuplicateViewMixin);
```

**Review by Tardo — CHANGES_REQUESTED** [review]

**Comment by chienandalu** [comment]
Thanks for your enlighting comments @Tardo :slightly_smiling_face: All your fixes applied

**Review by Tardo — APPROVED** [review]

**Pull request merged**